rocketchat-core/src/main/java/com/rocketchat/core/ChatRoom.java

```diff
@@ -155,10 +172,11 @@ public void setFavourite(Boolean isFavoutite, SimpleCallback callback) {
     //Subscription methods
 
     public void subscribeRoomMessageEvent(SubscribeListener subscribeListener,
```

Comment:
`SubscribeListener`, is actualy a callback. It is called once and then removed....

Listeners may have one or more events.

Callbacks always run either `onSuccess` or `onError` just once.

Reply:
Actually it is not a callback, it is part of stream, which shows if particular room is subscribed or not. It doesn't return any error for subscription. Boolean shows if it is successful or not.

Reply:
Actually it is implemented as a Callback.

```java
    public void processSubscriptionSuccess(JSONObject subObj) {
        if (subObj.optJSONArray("subs") != null) {
            String id = subObj.optJSONArray("subs").optString(0);
            if (listeners.containsKey(id)) {
                listeners.remove(id).onSubscribe(true, id);
            }
        }
    }
```

Reply:
Changed it to callback